linera-core/src/unit_tests/worker_tests.rs

```diff
@@ -340,9 +340,22 @@ where
         .take(block.operations.len())
         .collect();
     let state_hash = system_state.into_hash().await;
+    let previous_message_blocks = messages
```

Comment:
What if there was no message for `recipient` in the previous block? Shouldn't we put in the latest one, regardless of how far way it would be?